src/HelloTriangle.cpp

```diff
@@ -333,7 +346,12 @@ int helloTriangle() try{
 	std::tie( graphicsQueueFamily, presentQueueFamily ) = getQueueFamilies( physicalDevice, surface );
 
 	const VkPhysicalDeviceFeatures features = {}; // don't need any special feature for this demo
-	const vector<const char*> deviceExtensions = { VK_KHR_SWAPCHAIN_EXTENSION_NAME };
+	vector<const char*> deviceExtensions = { VK_KHR_SWAPCHAIN_EXTENSION_NAME };
+
+	// https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkDeviceCreateInfo.html#VUID-VkDeviceCreateInfo-pProperties-04451
+	if (isExtensionRequested(VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME, requestedInstanceExtensions)) {
+		deviceExtensions.push_back(VK_KHR_PORTABILITY_SUBSET_EXTENSION_NAME);

```

Comment:
Would having that code behind a `#ifdef VULKAN_PORTABILITY` work? I don't have a setup with both Portability & non-Portability drivers to test on. Thanks

Reply:
Nobody probably has. It is just a matter of the code being generic, rather than platform-specific.

What I think should be done here is enable `VK_KHR_portability_subset` extension if and only if the device supports it (via `vkEnumerateDeviceExtensionProperties`). And if it does not support it, then we can just take the device as is without enabling the extensions (otherwise enabling extension that is not supported would result in a `VK_ERROR_EXTENSION_NOT_PRESENT`).

Reply:
Thanks, wouldn't checking `isExtensionSupported(VK_KHR_PORTABILITY_SUBSET_EXTENSION_NAME, supportedInstanceExtensions)` achieve the same result?

Reply:
`VK_KHR_portability_subset` is a device extension. The `isExtensionSupported` would work, but needs `supportedDeviceExtensions` instead.